flytekit/annotated/condition.py

```diff
@@ -269,14 +269,19 @@ def else_(self) -> Case:
 }
 
 
+def create_branch_node_promise_var(node_id: str, var: str) -> str:
+    return f"{node_id}.{var}"
```

Comment:
Could you add a comment explaining why this is necessary?  Is it just in cases where there's an out_0 from two parents?

Reply:
true... will do